ipv8/REST/asyncio_endpoint.py

```diff
@@ -0,0 +1,240 @@
+            self.strategy = None
+            return True
+        return True
+
+    @docs(
+        tags=["Asyncio"],
+        summary="Measure the core drift.",
+        responses={
+            200: {
+                "schema": schema(DriftResponse={"measurements": [
+                    schema(Measurement={
+                        "timestamp": Float,
+                        "drift": Float
+                    })
+                ]})
+            },
+            400: {
+                "schema": DefaultResponseSchema,
+                "example": {"success": False, "error": "Core drift disabled."}
+            }
+        }
+    )
+    async def retrieve_drift(self, _):
+        if self.strategy is None:
+            return Response({"success": False, "error": "Core drift disabled."}, status=HTTP_NOT_FOUND)
+        with self.strategy.walk_lock:
+            return Response({"measurements": [{"timestamp": e[0], "drift": e[1]} for e in self.strategy.history]})
+
+    @docs(
+        tags=["Asyncio"],
+        summary="Enable or disable drift measurements.",
+        responses={
+            200: {
+                "schema": DefaultResponseSchema,
+                "example": {"success": True}
+            },
+            400: {
+                "schema": DefaultResponseSchema,
+                "examples": {'Enable value not specified': {"success": False, "error": "incorrect parameters"}}
+            }
+        }
+    )
+    @json_schema(schema(EnableDriftRequest={
+        'enable*': (Boolean, 'Whether to enable or disable measuring.'),
+    }))
+    async def enable_measurements(self, request):
+        parameters = await request.json()
+        if 'enable' not in parameters:
+            return Response({"error": "incorrect parameters"}, status=HTTP_BAD_REQUEST)
+
+        status = self.enable() if parameters.get('enable') else self.disable()
+        if status:
+            return Response({"success": True})
+        else:
+            return Response({"success": False, "error": "Session not initialized."})
+
+    @docs(
+        tags=["Asyncio"],
+        summary="Return all currently running Asyncio tasks.",
+        responses={
+            200: {
+                "schema": schema(AsyncioTasksResponse={"tasks": [
+                    schema(AsyncioTask={
+                        "name": String,
+                        "running": Boolean,
+                        "stack": [String],
+                        "taskmanager": String,
+                        "start_time": Float,
+                        "interval": Integer
+                    })
+                ]})
+            },
+            400: {
+                "schema": DefaultResponseSchema,
+                "example": {"success": False, "error": "at least Python 3.7 is required"}
+            }
+        }
+    )
+    async def get_asyncio_tasks(self, _):
+        if sys.version_info[1] < 7:
```

Comment:
This is a project requirement already, we don't support anything below Python 3.7. Misconfiguring the OS is HTTP 500 worthy i.m.o. (and saves adding this if-statement).

Reply:
It's there because some of our users are running Tribler under Python 3.6. Back when we released v7.5.0-exp2, we had several reports related to Python 3.6 compatibility.

However, if you're announcing that 3.6 is not supported, I'll stop trying to keep things compatible.

Reply:
Nevermind, I'll just remove it 🤷

Reply:
This is actually in the `setup.py`: https://github.com/Tribler/py-ipv8/blob/master/setup.py#L36

(which we should update to also include Python 3.8 now that I look at it)